src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -215,8 +218,8 @@ def _init_query_compiler(
         elif isinstance(data, Index):
             query_compiler = data._query_compiler
         else:
-            query_compiler = DataFrame(
-                index=cls._NATIVE_INDEX_TYPE(data=data, **kwargs)
+            query_compiler = from_pandas(
```

Comment:
Changed to prevent infinitely cycling between DF constructor and Index constructor.